pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -172,6 +176,9 @@ public static class Instance {
     public static final String CONFIG_OF_PINOT_BROKER_STARTABLE_CLASS = "pinot.broker.startable.class";
     public static final String CONFIG_OF_PINOT_SERVER_STARTABLE_CLASS = "pinot.server.startable.class";
     public static final String CONFIG_OF_PINOT_MINION_STARTABLE_CLASS = "pinot.minion.startable.class";
+
+    public static final String CONFIG_OF_MULTI_STAGE_ENGINE_ENABLED = "pinot.server.multi_stage.enabled";
```

Comment:
(nit) use `pinot.server.multistage.engine.enabled` ?

Reply:
I wonder where should we specify this config. May be both broker and server or a general config ?

The deployment / upgrade model is typically controller - broker - server. I think our expectation is that user who wants to try this new feature from master release will upgrade and then enable this config followed by restart. In that case may be a single config is fine. 

But let's say broker is upgraded / new and this feature flag is set to true but server is yet to be upgraded. What happens in this case if broker receives a query that needs to be sent to multistage engine ?

Reply:
the key is that both broker and server should shared the same `pinot.multistage.engine.enabled` config key (one config is enough). 

in the situation of upgrade, if broker is enabled but server is not (e.g. not yet upgraded) then broker query planner (QueryEnvironment class) cannot generate a runnable query plan, because it cannot assign segments/tableScan to the appropriate servers.

this is because, when multi-stage query engine on server is enabled; a `multi-stage.query.request.port` and `multi-stage.query.mailbox.port` is configured and registered in Zookeeper associated with the `ServerInstance` ZNode. if those ports are -1, broker cannot use them for runnable plan generation.